### PR TITLE
Remove DiffEqBase git magic

### PR DESCRIPTION
**Pull request by devmotion**
Fixes https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/issues/929.

### CODE REVIEW

**Comment by devmotion** [comment]
Tests fail since currently ModellingToolkit is broken: the call in https://github.com/JuliaDiffEq/ModelingToolkit.jl/blob/9b8e9c44db17f9cfa7acdca1d1acfd46e6514942/src/systems/diffeqs/diffeqsystem.jl#L237 causes an error in https://github.com/JuliaDiffEq/ModelingToolkit.jl/blob/9b8e9c44db17f9cfa7acdca1d1acfd46e6514942/src/systems/diffeqs/diffeqsystem.jl#L184.

**Comment by kanav99** [comment]
Can we do some kind of try-catch construct instead?

**Comment by devmotion** [comment]
I'm not sure if it's worth trying to fix the current implementation. IMO using a dedicated Manifest.toml file is the safer, more general, and more future-proof approach.

**Comment by devmotion** [comment]
Tests are still broken and require ModelingToolkit 0.7.2 with https://github.com/JuliaDiffEq/ModelingToolkit.jl/pull/179

**Comment by devmotion** [comment]
Tests pass now, so hopefully no additional regressions are present in ModelingToolkit and/or OrdinaryDiffEq.

File: test/interface/jacobian_tests.jl [hunk 1]
```diff
@@ -46,7 +46,7 @@ end
 
 prob = ODEProblem(lotka,[1.0,1.0],(0.0,1.0),[1.5,1.0,3.0,1.0])
 de = ModelingToolkit.modelingtoolkitize(prob)
-prob2 = remake(prob, f=ODEFunction(de..., Val(false), Wfact=true))
+prob2 = remake(prob, f=ODEFunction(de...; Wfact=true))
```
Comment (ChrisRackauckas) [inline test/interface/jacobian_tests.jl]: should add `jac=true`

**Comment by devmotion** [comment]
Gitlab errors are unrelated and should be fixed by https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/pull/932

**Comment by devmotion** [comment]
Any additional comments or suggestions?

**Comment by ChrisRackauckas** [comment]
nope looks good

**Pull request merged**